_episodes/10-prog.md

```diff
@@ -188,6 +188,44 @@ so that they are safe to use.
 > between 10.0 and 25.0.  How long does it take this program to run?
 > How long does it take to run a program that simply writes those
 > random numbers to a file?
+>
+> > ## Solution
+> > ~~~
+> > import sqlite3
+> > # import random number generator
+> > from numpy.random import uniform
+> >
+> > random_numbers = uniform(low=10.0, high=25.0, size=100000)
+> >
+> > connection = sqlite3.connect("original.db")
+> > cursor = connection.cursor()
+> > cursor.execute("CREATE TABLE Pressure (reading float not null)")
+> > query = "INSERT INTO Pressure values (?);"
+> >
+> > for number in random_numbers:
+> >     cursor.execute(query, [number])
```

Comment:
You can actually use [`executemany()`](https://docs.python.org/3.5/library/sqlite3.html#sqlite3.Cursor.executemany) here, which should be faster

Reply:
I would be careful with introducing too many new things at one. `executemany` + generator/list comprehensions might be too much at once. What do you say?

Reply:
This is not a programming course, so knowledge of Python is assumed. `executemany()` makes solution 1 and 3 simpler by removing the loop, and I think the generator is easily understandable for 2 even if you didn't encounter the syntax before (`cursor_backup.execute(query, (entry for entry in results if entry[0] > 20.0))`). Do you think it's really too advanced?

Reply:
List comprehensions are only covered in the [intermediate and advanced](
http://intermediate-and-advanced-software-carpentry.readthedocs.io/en/latest/idiomatic-python.html)
courses as far as I know. So I wouldn’t assume familiarity in a beginners
SQL workshop.